src/TwigComponent/src/Event/PreMountEvent.php

```diff
@@ -12,13 +12,14 @@
 namespace Symfony\UX\TwigComponent\Event;
 
 use Symfony\Contracts\EventDispatcher\Event;
+use Symfony\UX\TwigComponent\ComponentMetadata;
 
 /**
  * @author Ryan Weaver <ryan@symfonycasts.com>
  */
 final class PreMountEvent extends Event
 {
-    public function __construct(private object $component, private array $data)
+    public function __construct(private object $component, private array $data, private ComponentMetadata $metadata)
```

Comment:
Make this argument optional defaulting to null. If it's null, trigger a deprecation. Make the getter below have a nullable return type.